core/events/events_comment_resize.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Class for comment resize event.
+ *
+ * @class
+ */
+
+import * as registry from '../registry.js';
+import {Size} from '../utils/size.js';
+import type {WorkspaceComment} from '../comments/workspace_comment.js';
+
+import {CommentBase, CommentBaseJson} from './events_comment_base.js';
+import * as eventUtils from './utils.js';
+import type {Workspace} from '../workspace.js';
+
+/**
+ * Notifies listeners that a workspace comment has resized.
+ */
+export class CommentResize extends CommentBase {
+  override type = eventUtils.COMMENT_RESIZE;
+
+  /** The size of the comment before the resize. */
+  oldSize?: Size;
+
+  /** The size of the comment after the resize. */
+  newSize?: Size;
+
+  /**
+   * @param opt_comment The comment that is being resized. Undefined for a blank
+   *     event.
+   */
+  constructor(opt_comment?: WorkspaceComment) {
+    super(opt_comment);
+
+    if (!opt_comment) {
+      return; // Blank event to be populated by fromJson.
+    }
+
+    this.oldSize = opt_comment.getSize();
+  }
+
+  /**
+   * Record the comment's new size. Called after the resize. Can only be
+   * called once.
+   */
+  recordNew() {
+    if (this.newSize) {
+      throw Error(
+        'Tried to record the new size of a comment on the ' +
+          'same event twice.',
+      );
+    }
+    const workspace = this.getEventWorkspace_();
+    if (!this.commentId) {
+      throw new Error(
+        'The comment ID is undefined. Either pass a comment to ' +
+          'the constructor, or call fromJson',
+      );
+    }
+    const comment = workspace.getCommentById(this.commentId);
+    if (!comment) {
+      throw new Error(
+        'The comment associated with the comment resize event ' +
+          'could not be found',
+      );
+    }
+    this.newSize = comment.getSize();
+  }
+
+  /**
+   * Encode the event as JSON.
+   *
+   * @returns JSON representation.
+   */
+  override toJson(): CommentResizeJson {
+    const json = super.toJson() as CommentResizeJson;
+    if (!this.oldSize) {
+      throw new Error(
+        'The old comment size is undefined. Either pass a comment to ' +
+          'the constructor, or call fromJson',
+      );
+    }
+    if (!this.newSize) {
+      throw new Error(
+        'The new comment size is undefined. Either call recordNew, or ' +
+          'call fromJson',
+      );
+    }
+    json['oldSize'] =
+      `${Math.round(this.oldSize.width)}, ${Math.round(this.oldSize.height)}`;
+    json['newSize'] =
+      `${Math.round(this.newSize.width)}, ${Math.round(this.newSize.height)}`;
```

Comment:
Rather than serializing to string, it might be cleaner to just serialize the new/old width and height separately?

Reply:
Done! (This was modeled after CommentMove, which combined the coordinates, but I agree that separating them feels cleaner.)